src/scanners/common/fetchTags.ts

```diff
@@ -0,0 +1,41 @@
+import {RateLimiter, ResourceTags} from '@/types'
+import {GetResourcesCommand, ResourceGroupsTaggingAPIClient} from '@aws-sdk/client-resource-groups-tagging-api'
+
+export const fetchTags = async (
+	resourceARNList: string[],
+	rateLimiter: RateLimiter,
+): Promise<ResourceTags | undefined> => {
+	if (!resourceARNList.length) {
+		return undefined
+	}
+
+	const client = new ResourceGroupsTaggingAPIClient()
+
+	const tagResult: Record<string, Record<string, string | undefined>> = {}
+
+	try {
+		const command = new GetResourcesCommand({
+			ResourceARNList: resourceARNList,
+		})
+
+		const response = await rateLimiter.throttle(() => client.send(command))
```

Comment:
Could this command require pagination or does it return tags for all the resources in the resourceARNList at once?